### DISCUSSION TITLE
Working with HII from OS

### DISCUSSION THREAD

**Discussion by Kostr** [tianocore/edk2#3334] [Q&A]
UEFI specification states that there is a way to work with HII data from the OS (section "OS Runtime Utilization"), i.e. it is possible to
create a form browser application and read/modify platform HII settings.

As far as I understand it correctly for this firmware on the target platform need to:
- use ExportPackageLists() to export HII Database content into a runtime buffer,
- use ExportConfig() to export the current configuration into a runtime buffer,
- provide these buffers via UEFI Configuration Tables

Modifications from the OS in this case are handled via:
- GetVariable()/SetVariable() runtime services  (for efivarstore's)
- UEFI UpdateCapsule runtime service (for varstore's and namedvaluestore's)

Is there any known implementation of such functionality?
- Does any of the open-source platforms perform HII Database save (https://github.com/tianocore/edk2 / https://github.com/tianocore/edk2-platforms / ...) ?
- Are there any open-source OS apps that can work with HII Database content?

**Comment by changab** [comment]
We are interested in this topic and is there any follow-up discussion on this?
